src/wallet/adapters/ConnectAdaptor.ts

```diff
@@ -49,53 +49,117 @@ export class ConnectAdaptor implements AUTHAdapter {
+  }
+
+  async connect(walletAddress?: string): Promise<void> {
     this.projectParams = await this.API.getProjectParams()
 
     const isWebAuthnCompatible = await webAuthnService.isWebAuthnCompatible()
 
-    if (!isWebAuthnCompatible) {
-      this.signer = walletAddress
-        ? await burnerWalletService.getSigner(
-            this.API,
-            this.provider,
-            walletAddress
-          )
-        : await burnerWalletService.createSignerAndWallet(this.API)
+    if (walletAddress) {
+      await this._verifywalletAddress(walletAddress)
+
+      if (isWebAuthnCompatible) {
+        const { publicKeyId, signerAddress } = await webAuthnService.getSigner({
+          API: this.API,
+          walletAddress
+        })
+        this.signer = new WebAuthnSigner(publicKeyId, signerAddress)
+      } else {
+        this.signer = await burnerWalletService.getSigner({
+          API: this.API,
+          provider: this.provider,
+          walletAddress
+        })
+      }
+
+      this.walletAddress = walletAddress
     } else {
-      try {
-        const { publicKeyId, signerAddress } = walletAddress
-          ? await webAuthnService.getSigner(this.API, walletAddress)
-          : await webAuthnService.createSignerAndWallet(this.API, this.userName)
+      if (isWebAuthnCompatible) {
+        const {
+          publicKeyX,
+          publicKeyY,
+          publicKeyId,
+          signerAddress,
+          deviceData,
+          walletAddress
+        } = await webAuthnService.createSigner({
+          API: this.API,
+          userName: this.userName
+        })
 
         this.signer = new WebAuthnSigner(publicKeyId, signerAddress)
-      } catch {
-        throw new Error(
-          'New Domain detected. You need to add that domain as signer.'
-        )
+        this.walletAddress = walletAddress
+
+        await this.API.initWalletWithWebAuthn({
+          walletAddress,
+          publicKeyId,
+          publicKeyX,
+          publicKeyY,
+          deviceData
+        })
+      } else {
+        const { signer, walletAddress } =
+          await burnerWalletService.createSigner({
+            API: this.API
+          })
+
+        this.signer = signer
+        this.walletAddress = walletAddress
+
+        await this.API.initWallet({
+          ownerAddress: signer.address
+        })
       }
     }
-
-    this.walletAddress = await this._initAdaptorWalletAddress(walletAddress)
   }
 
-  async _verifywalletAddress(walletAddress: string): Promise<void> {
-    let connectWallet
-    try {
-      connectWallet = await this.API.getWalletInfos(walletAddress)
-    } catch {
-      throw new Error('Invalid address format')
+  async importSafe(
+    walletAddress: string,
+    message: string,
+    signature: string
+  ): Promise<string> {
+    let requestBody
+
+    const isWebAuthnCompatible = await webAuthnService.isWebAuthnCompatible()
+
+    if (!isWebAuthnCompatible) {
```

Comment:
You don't check that a wallet with this address alreadty exist ?
Even if its the API that verify, you can do it here too.